Modify base sequence regex

teemukataja opened the pull request:
Fixes #221 

PR to fix a forgotten issue.

sdelatorrep commented:
Please @teemukataja, update the beacon.md so it reflects this change. Also, I think we should explain what this change means (e.g. use N as a wildcard: `ACNCCNN`) in both the beacon.md and the description of the fields in beacon.yaml.
I think we shouldn't merge this PR until this is done.

teemukataja commented:
> Please @teemukataja, update the beacon.md so it reflects this change. Also, I think we should explain what this change means (e.g. use N as a wildcard: ACNCCNN) in both the beacon.md and the description of the fields in beacon.yaml.

Rebased and added explanation to description.

sdelatorrep commented on beacon.md:
```diff
@@ -255,15 +255,15 @@ An example `GET` request and response to the info endpoint:
 |Parameter (* mandatory, ~ one of these is mandatory)|Description|Type|Example|
 |---|---|:---:|---|
 |*referenceName**|Reference name (chromosome). Accepting values 1-22, X, Y so follows Ensembl chromosome naming convention.|string|`'1'`|
-|*referenceBases**|Reference bases for this variant (starting from `start`). Accepted values: [ACGT]* When querying for variants without specific base alterations (e.g. imprecise structural variants with separate `variantType` as well as `startMin` & `endMin` ... parameters), the use of a single "N" value is required.<br/>See the REF field in [VCF 4.2 specification](https://samtools.github.io/hts ...
```
@teemukataja , any reason to remove the reference to the VCF specification? You kept it in the `alternateBases` description but not here.

teemukataja replied:
No, must have been removed accidentally.

teemukataja commented on beacon.md:
```diff
@@ -255,15 +255,15 @@ An example `GET` request and response to the info endpoint:
 |---|---|:---:|---|
 |*referenceName**|Reference name (chromosome). Accepting values 1-22, X, Y so follows Ensembl chromosome naming convention.|string|`'1'`|
-|*referenceBases**|Reference bases for this variant (starting from `start`). Accepted values: [ACGT]* When querying for variants without specific base alterations (e.g. imprecise structural variants with separate `variantType` as well as `startMin` & `endMin` ... parameters), the use of a single "N" value is required.<br/>See the REF field in [VCF 4.2 specification](https://samtools.github.io/hts ...
+|*referenceBases**|Reference bases for this variant (starting from `start`). Accepted values: [ACGTN]*. N is a wildcard, that denotes the position of any base, and can be used as a standalone base of any type or within a partially known sequence. For example a sequence where the first and last bases are known, but the middle portion can exhibit countless variations of [ACGT], or the bases are unk ...
```
```suggestion
|*referenceBases**|Reference bases for this variant (starting from `start`). Accepted values: [ACGTN]*. N is a wildcard, that denotes the position of any base, and can be used as a standalone base of any type or within a partially known sequence. For example a sequence where the first and last bases are known, but the middle portion can exhibit countless variations of [ACGT], or the bases are unknown: ANNT the Ns can take take any form of [ACGT], which makes both ACCT and ATGT (or any other combination) viable sequences.<br/>See the REF field in [VCF 4.2 specification](https://samtools.github.io/hts-specs/VCFv4.2.pdf).|string|`'G'`|
```

mbaudis commented:
@sdelatorrep @teemukataja Independent of the (correct) fix here: How about patterns w/ a range of "N" bases: `/AC(?:N{1,3})TT/` ? While I don't think about providing patterns, a limited amount of alternative codings in the request could be of interest...
I've stumbled on a problem here when doing wildcard matches, where one usually would want a `*` in the `alternateBases` query; `N` returns only the single base replacements (you can test this on Beacon+ w/ using `N`, `NN` ...).
